frame/society/src/lib.rs

```diff
@@ -1299,8 +1306,8 @@ impl<T: Trait<I>, I: Instance> Module<T, I> {
 				// select one as primary, randomly chosen from the accepted, weighted by approvals. 
 				// Choose a random number between 0 and `total_approvals`
 				let primary_point = pick_usize(&mut rng, total_approvals - 1);
-				// Find the user who falls on that point
-				let primary = accepted.iter().find(|e| e.1 > primary_point)
+				// Find the zero bid or the user who falls on that point
+				let primary = accepted.iter().find(|e| e.2.is_zero() || e.1 > primary_point)
```

Comment:
will the zero bid always be the first in the list? if not then the primary point will be chosen if it comes before the zero bidder.

Reply:
Yes. We must assume `bids` is sorted by value. If so, then `candidates` will be sorted by value. If so, then `accepted` will be sorted by value.

If you would like me to write this assumption down near this code I can do that.